plugins/inputs/cloud_pubsub/cloud_pubsub.go

```diff
@@ -90,6 +94,15 @@ func (ps *PubSub) Start(ac telegraf.Accumulator) error {
 		return fmt.Errorf(`"project" is required`)
 	}
 
+	if ps.MaxDecompressionSize <= 0 {
+		ps.MaxDecompressionSize = internal.DefaultMaxDecompressionSize
+	}
+
+	var err error
+	ps.decoder, err = internal.NewContentDecoder(ps.ContentEncoding)
```

Comment:
It would be nice if we checked the value of ContentEncoding in cloud_pubsub's init function. As it only supports two of the various options that telegraf provides I'd like to protect against odd behavior or errors if someone said "zlib" for example.